packages/twentytwenty-theme/src/components/archive/infinite-scroll-pagination.js

```diff
@@ -0,0 +1,91 @@
+import React, { useEffect, useState } from "react";
+import { connect, styled } from "frontity";
+import { useInView } from "react-intersection-observer";
+
+import PostSeparator from "../post/post-separator";
+import Loading from "../loading";
+import Archive from "./archive";
+
+const InfiniteScroller = ({
+  state,
+  actions,
+  libraries,
+  previousPageVisible,
+  currentLink
+}) => {
+  // Get the total posts to be displayed based for the current link
+  const { totalPages } = state.source.get(currentLink);
+  const { path, page, query } = libraries.source.parse(currentLink);
+
+  // Check if we can go to next page within the pagination
+  const isThereNextPage = page < totalPages;
+
+  // Get the link for the next page
+  const nextPageLink = libraries.source.stringify({
+    path,
+    page: page + 1,
+    query
+  });
+
+  // Create a ref to ScrollObserver and observe it's visiblility.
+  // ScrollObserver is a "hit area" - as soon as it's visible in the viewport
+  const [scrollObserverRef, scrollObserverVisible] = useInView();
```

Comment:
I thought about this, but this means that we would have 2 refs:
a) `scrollObserverRef` (with 500px rootMargin) and  
b) `nextPageRef` (the same as we have now)
And then we pass BOTH of them to `<ArchiveContainer />` 
Since you cannot pass 2 refs to the same component, we'd have to use something like: https://github.com/seznam/compose-react-refs, which I think is an overkill for the example.